x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/ReplaceAggregateNestedExpressionWithEval.java

```diff
@@ -159,10 +125,73 @@ protected LogicalPlan rule(Aggregate aggregate) {
             aggregate = aggregate.with(newEval, groupings, aggregates);
         }
 
-        return (LogicalPlan) aggregate;
+        return aggregate;
+    }
+
+    private static Expression transformNonEvaluatableGroupingFunction(
+        GroupingFunction.NonEvaluatableGroupingFunction gf,
+        List<Alias> evals
+    ) {
+        int[] counter = new int[] { 0 };
```

Comment:
I think we don't need this wrapped here (?)

Reply:
Correct, 🙏 .